src/earn/saga.ts

```diff
@@ -338,77 +338,72 @@ export function* withdrawSubmitSaga(action: PayloadAction<WithdrawInfo>) {
       `${TAG}/withdrawSubmitSaga`,
-      `Starting withdraw for token ${tokenId}, total transactions: ${preparedTransactions.length}`
+      `Starting ${mode} for token ${tokenId}, total transactions: ${preparedTransactions.length}`
     )
 
-    const createWithdrawStandbyTxHandlers = []
+    const transactionHandlers: Array<
+      (transactionHash: string, feeCurrencyId?: string) => BaseStandbyTransaction
+    > = []
 
-    // Assumes the first tx is withdraw and the rest are claim rewards
-    const createWithdrawStandbyTxHandler = (
-      transactionHash: string,
-      feeCurrencyId?: string
-    ): BaseStandbyTransaction => {
-      return {
+    if (mode !== 'claim-rewards') {
+      const createWithdrawStandbyTxHandler = (
+        transactionHash: string,
+        feeCurrencyId?: string
+      ): BaseStandbyTransaction => ({
         context: newTransactionContext(TAG, 'Earn/Withdraw'),
         networkId,
         type: TokenTransactionTypeV2.EarnWithdraw,
-        inAmount: {
-          value: amount ?? pool.balance,
-          tokenId,
-        },
-        outAmount: {
-          value: amount ?? pool.balance,
-          tokenId: pool.dataProps.withdrawTokenId,
-        },
+        inAmount: { value: amount ?? pool.balance, tokenId },
+        outAmount: { value: amount ?? pool.balance, tokenId: pool.dataProps.withdrawTokenId },
         transactionHash,
         feeCurrencyId,
         providerId: pool.appId,
-      }
+      })
+      transactionHandlers.push(createWithdrawStandbyTxHandler)
     }
 
-    createWithdrawStandbyTxHandlers.push(createWithdrawStandbyTxHandler)
-
     rewardsTokens.forEach(({ balance, tokenId }, index) => {
       const createClaimRewardStandbyTx = (
         transactionHash: string,
         feeCurrencyId?: string
-      ): BaseStandbyTransaction => {
-        return {
-          context: newTransactionContext(TAG, `Earn/ClaimReward-${index + 1}`),
-          networkId,
-          amount: {
-            value: balance,
-            tokenId,
-          },
-          type: TokenTransactionTypeV2.EarnClaimReward,
-          transactionHash,
-          feeCurrencyId,
-          providerId: pool.appId,
-        }
+      ): BaseStandbyTransaction => ({
+        context: newTransactionContext(TAG, `Earn/ClaimReward-${index + 1}`),
+        networkId,
+        amount: { value: balance, tokenId },
+        type: TokenTransactionTypeV2.EarnClaimReward,
+        transactionHash,
+        feeCurrencyId,
+        providerId: pool.appId,
+      })
+
+      if (mode === 'claim-rewards' || !pool.dataProps.withdrawalIncludesClaim) {
+        transactionHandlers.push(createClaimRewardStandbyTx)
       }
-      createWithdrawStandbyTxHandlers.push(createClaimRewardStandbyTx)
     })
 
-    AppAnalytics.track(EarnEvents.earn_withdraw_submit_start, commonAnalyticsProps)
+    const eventStart =
+      mode === 'claim-rewards'
+        ? EarnEvents.earn_claim_submit_start
+        : EarnEvents.earn_withdraw_submit_start
```

Comment:
Optional nit: I think it would be cleaner to keep just one event, earn_withdraw_submit_start, and add `mode` as one of the properties.

Reply:
I believe it’s best to keep the existing structure intact. Changing to a shared analytic event would break existing analytic tracking for the Earn:  [Earn Multi Pool Dashboard](https://mixpanel.com/project/2786968/view/3321744/app/boards#id=8096545).

Reply:
Adjusted analytics in 0b062d8813cf11f69882255ca1d5b1b5d59c44bd!